vignettes/articles/multi-scale-resampling.Rmd

```diff
@@ -0,0 +1,233 @@
+  group_by(PLOT) |> 
+  summarise(
+    yr = mean(INVYR, na.rm = TRUE),
+    plot = mean(PLOT, na.rm = TRUE),
+    lat = mean(LAT, na.rm = TRUE),
+    long = mean(LON, na.rm = TRUE),
+    n_trees = n(),
+    agb = sum(DRYBIO_AG, na.rm = TRUE)
+  ) |> 
+  collect() |> 
+  mutate(
+    # Because of how we joined, `n_trees` is always at least 1 -- 
+    # even if there are 0 trees
+    n_trees = ifelse(is.na(agb) & n_trees == 1, 0, n_trees),
+    agb = ifelse(is.na(agb), 0, agb)
+  )
+
+DBI::dbDisconnect(con)
+
+readr::write_csv(plot_measurements, "plot_measurements.csv")
+```
+
+</details>
+
+With that pre-processing done, it's time for us to load our data and turn it into an sf object. We're going to reproject our data to use a coordinate reference system that the US government tends to use for national data products, like the FIA:
+
+```{r}
+#| message: false
+library(sf)
+
+invisible(sf_proj_network(TRUE))
+
+plot_measurements <- readr::read_csv("https://www.mm218.dev/junkdrawer/riemann/plot_measurements.csv") |> 
+  st_as_sf(coords = c("long", "lat"), crs = 4326) |> 
+  st_transform(5070)
+```
+
+And this is what we're going to go ahead and resample. We want to assess our model's performance at multiple scales, following the approach in Riemann et al. That means we need to do the following:
+
+1. Block our study area using multiple sets of regular hexagons of different sizes, and assign our data to the hexagon it falls into within each set.
+2. Perform leave-one-block-out cross-validation with each of those sets, fitting our model to n - 1 of the n hexagons we've created and assessing it on the hold-out hexagon.
+3. Calculate model accuracy for each size based on those held-out hexes.
+
+So to get started, we need to block our study area. We can do this using the `spatial_block_cv()` function from spatialsample. We'll generate ten different sets of hexagon tiles, using `cellsize` arguments of between 10,000 and 100,000 meters[^3]. The code to do that, and to store all of our resamples in a single tibble, looks like this[^4]:
+
+[^3]: This value is in meters because our coordinate reference system is in meters. It represents the length of the [apothem](https://en.wikipedia.org/wiki/Apothem), from the center of each polygon to the middle of the side. We're using hexagons because Riemann et al. also used hexagons. 
+
+[^4]: `v` is `Inf` because we want to perform leave-one-block-out cross-validation, but we don't know how many blocks there will be before they're created. This is the supported way to do leave-one-X-out cross-validation in spatialsample > 0.2.0 (another option is to set `v = NULL`).
+
+```{r, message=FALSE}
+set.seed(123)
+library(dplyr)
+library(spatialsample)
+cellsize <- seq(10, 100, 10) * 1000
+
+riemann_resamples <- tibble(
+  resamples = purrr::map(
+    cellsize, 
+    \(cs) {
+      spatial_block_cv(
+        plot_measurements,
+        v = Inf,
+        cellsize = cs,
+        square = FALSE
+      )
```

Comment:
I think, for readability, pull out this anonymous function and give it a name, like `resample_at_cellsize()` or something like that.